pkg/controller/migration/convert/convert.go

```diff
@@ -103,11 +94,33 @@ func getComponents(ctx context.Context, client client.Client) (*components, erro
 		typha:           t,
 	}
 
-	err := loadCNI(comps)
+	// do some upfront processing of CNI by loading it into comps
+	var err error
+	comps.cni, err = loadCNI(comps)
 
 	return comps, err
 }
 
+// loadCNI pulls the CNI network config from it's env var source within components
+// and then returns the parsed data.
+func loadCNI(comps *components) (nc cni.NetworkComponents, err error) {
+	// do some upfront processing of CNI by loading it into comps
+	c := getContainer(comps.node.Spec.Template.Spec, containerInstallCNI)
+	if c == nil {
+		return
```

Comment:
In addition to the logs, I was also suggesting that we should keep the comment that no install cni container is allowed because it is not expected for non-Calico CNI. I don't think that is necessary though.